lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,278 @@
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_RIGHT]: 'below right',
+	[DATE_PICKER_CALENDAR_POSITIONS.BOTTOM]: 'below',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_LEFT]: 'above left',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP_RIGHT]: 'above right',
+	[DATE_PICKER_CALENDAR_POSITIONS.TOP]: 'above',
+};
+export default {
+	name: 'DatePicker',
+	components: {
+		DatePickerBox,
+		DsTile,
+	},
+	props: {
+		triggerType: {
+			type: String as PropType<DatePickerTriggerTypes>,
+			default: DATE_PICKER_TRIGGER_TYPES.TILE,
+		},
+		isInteractive: {
+			type: Boolean,
+			default: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wybierz datę',
+		},
+		date: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		label: {
+			type: String,
+			default: '',
+		},
+		isLabelUppercase: {
+			type: Boolean,
+			default: false,
+		},
+		icon: {
+			type: Object,
+			default: null,
+			validate: (icon) => icon === null || Object.values(ICONS).includes(toRaw(icon)),
+		},
+		isIconHiddenOnMobile: {
+			type: Boolean,
+			default: false,
+		},
+		calendarPosition: {
+			type: String as PropType<DatePickerCalendarPositions>,
+			default: DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT,
+		},
+		errorMessage: {
+			type: String,
+			default: '',
+		},
+		state: {
+			type: (String as PropType<DatePickerStates>) || null,
+			default: DATE_PICKER_STATES.DEFAULT,
+		},
+		color: {
+			type: String as PropType<DatePickerColors>,
+			default: DATE_PICKER_COLORS.NEUTRAL,
+		},
+		defaultDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		disableDates: {
+			type: Array,
+			default: () => [],
+		},
+		minDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		maxDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+	},
+	emits: { 'update:date': () => true },
+	data() {
+		return {
+			datePicker: null,
+			isOpen: false,
+			DATE_PICKER_STATES: Object.freeze(DATE_PICKER_STATES),
+			DATE_PICKER_TRIGGER_TYPES: Object.freeze(DATE_PICKER_TRIGGER_TYPES),
+		};
+	},
+	computed: {
+		isDatePickerEnabled() {
+			return this.isInteractive && this.state === DATE_PICKER_STATES.DEFAULT;
+		},
+		eyebrowText() {
+			if (this.state === DATE_PICKER_STATES.LOADING) {
+				return '';
+			}
+
+			return capitalizeFirstLetter(
+				new Date(this.date).toLocaleDateString(undefined, {
+					dateStyle: undefined,
+					timeStyle: undefined,
+					weekday: 'long',
+				}),
+			);
+		},
+		text() {
+			if (this.state === DATE_PICKER_STATES.LOADING || !this.date.length) {
+				return this.placeholder;
+			}
+
+			return new Date(this.date).toLocaleDateString(undefined, {
+				dateStyle: 'medium',
+				timeStyle: undefined,
+			});
+		},
+	},
+	async mounted() {
+		if (this.isDatePickerEnabled && !this.datePicker) {
+			if (!flatpickr) {
+				const { default: flatpickrFunction } = await import('flatpickr');
+				flatpickr = flatpickrFunction;
+			}
+
+			this.datePicker = flatpickr(this.$refs.flatpickrInput, {
```

Comment:
![Zrzut ekranu 2024-06-28 o 09 51 35](https://github.com/bethinkpl/design-system/assets/4587534/3d513993-301c-4792-b1aa-e07e40cb74a7)
I think we're missing translations